### PR TITLE
add support for api communication on the client side

### PR DESCRIPTION
**Pull request by lucbic**

### CODE REVIEW

File: client/src/components/TodoList.vue [hunk 1]
```diff
@@ -9,27 +9,53 @@
...
   components: {
     Todo,
   },
   methods: {
     deleteTodo(todo) {
-      const todoIndex = this.todos.indexOf(todo);
-      this.todos.splice(todoIndex, 1);
+      axios.delete(`${url}/todo/${todo.id}`)
+      .then((response) => {
+        if (response.data.success === 'ok') {
```
Comment (badosu) [inline client/src/components/TodoList.vue]: This step is not necessary, if the response was a `2**` status code then the deletion was successful, you can change the server response to just `halt 204`

File: client/src/App.vue [hunk 2]
```diff
@@ -17,35 +20,42 @@ export default {
...
+        this.todos = response.data;
+      })
+      .catch((error) => {
+        this.status = `an error ocurred: ${error}`;
       });
     },
   },
 };
+
+// this.getTodo();
```
Comment (badosu) [inline client/src/App.vue]: You can remove this comment

File: client/package.json [hunk 3]
```diff
@@ -14,7 +14,9 @@
     "build": "node build/build.js"
   },
   "dependencies": {
+    "axios": "^0.17.1",
     "express": "^4.16.2",
+    "json-loader": "^0.5.7",
```
Comment (badosu) [inline client/package.json]: Why do you need this dependency?

File: client/src/components/TodoList.vue [hunk 4]
```diff
@@ -9,27 +9,53 @@
...
+      todoToggle.done = !todo.done;
+      axios.patch(`${url}/todo/${todo.id}`, todoToggle)
+      .then(() => {
+        //
+      })
+      .catch((error) => {
+        this.status = `an error ocurred: ${error}`;
+      });
+    },
+    editTodo(todo) {
```
Comment (badosu) [inline client/src/components/TodoList.vue:49]: It's better to keep the server update logic to the single todo component, as it's the responsability of it

File: client/src/App.vue [hunk 5]
```diff
@@ -17,35 +20,42 @@ export default {
...
-      }, {
-        title: 'Todo C',
-        project: 'Projeto C',
-        done: false,
-      }, {
-        title: 'Todo D',
-        project: 'Projeto D',
-        done: false,
-      }],
+      todos: this.getTodo(),
```
Comment (badosu) [inline client/src/App.vue]: Please rename to `getTodos()`

File: client/package.json [hunk 6]
```diff
@@ -14,7 +14,9 @@
     "build": "node build/build.js"
   },
   "dependencies": {
+    "axios": "^0.17.1",
     "express": "^4.16.2",
```
Comment (badosu) [inline client/package.json]: Also, this one should not be necessary as well.

File: client/src/components/TodoList.vue (hunk 4 above)
Comment (lucbic) [reply client/src/components/TodoList.vue:49]: I just maintained the logic of the original to-do Vue tutorial. The edit, toggle and delete functions were all implemented on the TodoList component

File: client/package.json (hunk 6 above)
Comment (lucbic) [reply client/package.json]: ok

File: client/src/App.vue (hunk 5 above)
Comment (lucbic) [reply client/src/App.vue]: ok

File: client/package.json (hunk 3 above)
Comment (lucbic) [reply client/package.json]: ok

File: client/src/App.vue (hunk 2 above)
Comment (lucbic) [reply client/src/App.vue]: ok

File: client/src/components/TodoList.vue (hunk 1 above)
Comment (lucbic) [reply client/src/components/TodoList.vue]: is the unnecessary step the `if` response check?

Comment (lucbic) [reply client/src/components/TodoList.vue]: ok

File: client/src/components/TodoList.vue (hunk 4 above)
Comment (lucbic) [reply client/src/components/TodoList.vue:49]: ok

**Comment by badosu** [comment]
LGTM 👍

**Pull request merged**